### PR TITLE
remove Trinity dependency/move crypto code to hmt-escrow

### LINKED ISSUE #272: Remove dependency on the `trinity` package
We have the package only for its dependencies, not for the package itself. What we actually use is `web3==5.12.0` and `py-evm==0.3.0a18`.
Since the `trinity` package is not maintained and its repository archived, there are no benefits of keep using it and it can be replaced with the real dependencies.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200560623420491/1201175417135913) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by adam-mrozik**
┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200560623420491/1201278021276808) by [Unito](https://www.unito.io)
Closes #272

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -6,14 +6,18 @@ verify_ssl = true
 [dev-packages]
 
 [packages]
-trinity = {git = "https://github.com/ethereum/trinity.git", ref = "master"}
-py-solc-x = {git = "https://github.com/iamdefinitelyahuman/py-solc-x.git", ref = "master"}
 boto3 = "*"
 hmt-basemodels = "*"
-Sphinx = {git = "https://github.com/sphinx-doc/sphinx.git", ref = "master"}
+web3 = "==5.12.0"
```
Comment (e271828-) [inline Pipfile]: Is there a reason to pin this to a point release?

File: hmt_escrow/crypto.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
+import os
```
Comment (e271828-) [inline hmt_escrow/crypto.py:24]: Where did this file come from? Please document source + license at the top. I'd guess it's MIT from a quick search on the strings, so looks fine, but needs to be recorded.

Comment (adam-mrozik) [reply hmt_escrow/crypto.py:24]: Added license + source

File: Pipfile (hunk 1 above)
Comment (adam-mrozik) [reply Pipfile]: Hmm, it was suggested in #272 , but tests work anyway, so I will leave it unpinned

File: Pipfile [hunk 3]
```diff
@@ -6,14 +6,15 @@ verify_ssl = true
...
+py-solc-x = "*"
+sphinx = "*"
+web3 = "*"
 
 [requires]
 python_version = "3.8"
 
 [pipenv]
-allow_prereleases = true
+allow_prereleases = true
```
Comment (spirit-q2) [inline Pipfile]:
```suggestion
allow_prereleases = true

```

**Review by spirit-q2 — COMMENTED** [review]
a small comment :)

**Comment by adam-mrozik** [comment]
@e271828- 
Please advise me if this is correct and if can be approved :)

**Review by e271828- — APPROVED** [review]

**Pull request merged**